aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -303,45 +356,41 @@ Tensor& mvlgamma_(Tensor& self, int64_t p) {
 // NB: If you use this macro, you may also need to add a CUDA forwarding
 // stub in CUDAUnaryOps
 
-#define IMPLEMENT_UNARY_OP_CORE(op)                                    \
+#define IMPLEMENT_UNARY_FLOATING_UFUNC_OP_CORE(op)                     \
   Tensor op(const Tensor& self) {                                      \
-    Tensor result = at::empty({0}, self.options());                    \
-    at::op##_out(result, self);                                        \
-    return result;                                                     \
+    return unary_floating_ufunc_cast_op_impl(self, at::op##_out);      \
```

Comment:
This still needs to be updated to cast if not CPU. The query should be if (CPU) ... else ...

Reply:
Done! https://github.com/pytorch/pytorch/pull/33322/commits/71e23c673e902017ab6d86445ace11bb1007144e